libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -17,7 +22,28 @@ ReturnCode CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element
         return ReturnCode::FAILURE;
     }
 
-    result.first = settingKey;
-    result.second = settingValue;
+    log_debug() << "key : " << settingKey << " value : " << settingValue;
+
+    if ("memory.limit_in_bytes" == settingKey) {
+        if (m_settings.count(settingKey)) {
+            // if the new value is lexicographical greater than old one set is as the value
+            if (std::stoi(m_settings[settingKey]) < std::stoi(settingValue)) {
+                m_settings[settingKey] = settingValue;
+            }
+        } else {
+            m_settings[settingKey] = settingValue;
+        }
+    } else {
+        log_warning() << "It is possible to apply this rule through control groups of working "
```

Comment:
But kernel can support this setting so maybe it is not told that its not supported. But we can keep saying this setting is not guaranteed to work by cgroup gateway. But i agree we should mentioned about the key. Thus i propose to keep sentence but changing "this rule" part with the actual setting key.